### PR TITLE
server: add log for binary execute statement

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
The dispatch error will print the packet data when meets some errors, e.g.
```
2018/10/22 15:45:15.528 conn.go:488: [warning] con:1 dispatch error:
id:1, addr:127.0.0.1:56631 status:2, collation:utf8_general_ci, user:root
"\x01\x00\x00\x00\x00\x01\x00\x00\x00\x01\x01\x06\x00"
[table:1048]Column 'i' cannot be null
```
It is not readable.

### What is changed and how it works?
Add additional messages for binary execute statement to make the user easy to know the executing SQL and its parameters.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
Run the following function though binary APIs.
```go
func errMsg(db *sql.DB) {
	var err error
	_, err = db.Exec("drop table if exists t")
	if err != nil {
		fmt.Println(err)
	}
	_, err = db.Exec("create table t (i int not null)")
	if err != nil {
		fmt.Println(err)
	}
	_, err = db.Exec("insert into t values (?)", nil)
	if err != nil {
		fmt.Println(err)
	}
}
```
it will add the addtional info, e.g.
```
insert into t values (?) [arguments: <nil>]
```

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Need to cherry-pick to release branches.

PTAL @lysu @tiancaiamao

### CODE REVIEW

**Comment by zz-jason** [comment]
@XuHuaiyu @eurekaka @lysu PTAL

**Comment by zz-jason** [comment]
@jackysp Do we need to cherry pick it to the **release-2.0** and **release-2.1** branch?

**Comment by jackysp** [comment]
Yes. I'll cherry-pick it after it merged. @zz-jason

File: server/conn_stmt.go [hunk 1]
```diff
@@ -564,3 +564,11 @@ func (cc *clientConn) handleSetOption(data []byte) (err error) {
 
 	return errors.Trace(cc.flush())
 }
+
+func (cc *clientConn) preparedStmt2String(stmtID uint32) string {
```
Comment (XuHuaiyu) [inline server/conn_stmt.go]: Should we add a comment for this function if we only use it for errorMsg
or
s/ preparedStmt2String/ preparedStmt2ErrMsg

Comment (jackysp) [reply server/conn_stmt.go]: I think we the function name should describe what it does. This function could also be used in other features, e.g. `show processlist`, so I think it is not necessary to change the name.

**Pull request merged**